app/grandchallenge/reader_studies/views.py

```diff
@@ -577,6 +622,29 @@ def get_success_url(self):
         return self.reader_study.get_absolute_url()
 
 
+class AddDisplaySetsToReaderStudy(AddObjectToReaderStudyMixin):
+    model = RawImageUploadSession
+    form_class = AddCasesForm
+    template_name = "reader_studies/readerstudy_add_object.html"
+    type_to_add = "images"
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs.update(
+            {
+                "user": self.request.user,
+                "linked_task": create_display_sets_for_upload_session.signature(
+                    kwargs={"reader_study_pk": self.reader_study.pk},
+                    immutable=True,
+                ),
+            }
+        )
+        return kwargs
+
+    def get_success_url(self):
+        return self.reader_study.get_absolute_url()
```

Comment:
Isn't it better to go to the upload session detail page?

Reply:
It is, I talked about this with Chris yesterday.